garden-backend-service/src/api/tasks/auto_deletion/utils.py

```diff
@@ -49,13 +91,51 @@ async def mark_gardens_for_deletion(session_maker):
 
 
 async def mark_modal_apps_for_deletion(session_maker) -> int:
-    # First get all modal function IDs that are used in gardens
-    used_function_ids = select(gardens_modal_functions.c.modal_function_id)
+    # First get all modal function IDs that are used in published gardens
+    used_function_ids = (
+        select(gardens_modal_functions.c.modal_function_id)
+        .join(Garden, Garden.id == gardens_modal_functions.c.garden_id)
+        .where(Garden.doi_is_draft.isnot(True))
+    )
 
-    # Then find modal apps where none of their functions are used
+    # Then find modal apps where none of their functions are used in the publised gardens
     stmt = select(ModalApp).where(
         ~ModalApp.modal_functions.any(ModalFunction.id.in_(used_function_ids))
     )
 
     async with session_maker() as db:
         return await _update_db_records_with_mark(db, stmt)
+
+
+async def unmark_marked_gardens(session_maker) -> int:
+    stmt = select(Garden).where(Garden.marked_for_deletion.isnot(None))
+    count = 0
+    async with session_maker() as db:
+        results = await db.scalars(stmt)
+        marked_gardens = results.all()
+        for g in marked_gardens:
+            # unmark marked gardens that have been published since they were marked
+            if not g.doi_is_draft:
+                g.marked_for_deletion = None
+                count += 1
+        await db.commit()
+    return count
+
+
+async def unmark_marked_modal_apps(session_maker) -> int:
+    # Subquery for all used modal function IDs
+    used_function_ids = select(gardens_modal_functions.c.modal_function_id)
```

Comment:
And it seems like the unmark operation isn't symmetrical. (I might be missing something but it looks like the marking logic would mark an app for deletion if it's used in an unpublished garden, and then this unmarking logic would unmark it right after. I don't see this conditioning on the state of the garden)